test/Microsoft.ML.Benchmarks/Harness/ProjectGenerator.cs

```diff
@@ -18,37 +18,39 @@ namespace Microsoft.ML.Benchmarks.Harness
+    /// this is why this class exists: 
+    ///  1. to tell MSBuild to copy the native dependencies to folder with .exe (NativeAssemblyReference)
+    ///  2. to generate a .csproj file that does not exclude Directory.Build.props (default BDN behaviour) which contains custom NuGet feeds that are required for restore step
     /// </summary>
     public class ProjectGenerator : CsProjGenerator
     {
         public ProjectGenerator(string targetFrameworkMoniker) : base(targetFrameworkMoniker, platform => platform.ToConfig(), null)
         {
         }
 
-        protected override void CopyAllRequiredFiles(ArtifactsPaths artifactsPaths)
-        {
-            base.CopyAllRequiredFiles(artifactsPaths);
-
-            CopyMissingNativeDependencies(artifactsPaths);
-        }
-
-        private void CopyMissingNativeDependencies(ArtifactsPaths artifactsPaths)
-        {
-            var foldeWithAutogeneratedExe = Path.GetDirectoryName(artifactsPaths.ExecutablePath);
-            var folderWithNativeDependencies = Path.GetDirectoryName(typeof(ProjectGenerator).Assembly.Location);
-
-            foreach (var nativeDependency in Directory
-                .EnumerateFiles(folderWithNativeDependencies)
-                .Where(fileName => ContainsWithIgnoreCase(fileName, "native")))
-            {
-                File.Copy(
-                    sourceFileName: nativeDependency,
-                    destFileName: Path.Combine(foldeWithAutogeneratedExe, Path.GetFileName(nativeDependency)), 
-                    overwrite: true);
-            }
-        }
-
-        bool ContainsWithIgnoreCase(string text, string word) => text != null && text.IndexOf(word, StringComparison.InvariantCultureIgnoreCase) >= 0;
+        protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+            => File.WriteAllText(artifactsPaths.ProjectFilePath, $@"
+<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <OutputPath>bin\{buildPartition.BuildConfiguration}</OutputPath>
+    <TargetFramework>{TargetFrameworkMoniker}</TargetFramework>
+    <AssemblyName>{artifactsPaths.ProgramName}</AssemblyName>
+    <AssemblyTitle>{artifactsPaths.ProgramName}</AssemblyTitle>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
+    <DebugType>pdbonly</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+  {GetRuntimeSettings(buildPartition.RepresentativeBenchmarkCase.Job.Environment.Gc, buildPartition.Resolver)}
+  <ItemGroup>
+    <Compile Include=""{Path.GetFileName(artifactsPaths.ProgramCodePath)}"" Exclude=""bin\**;obj\**;**\*.xproj;packages\**"" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include=""{GetProjectFilePath(buildPartition.RepresentativeBenchmarkCase.Descriptor.Type, logger).FullName}"" />
+    <NativeAssemblyReference Include=""CpuMathNative"" />
```

Comment:
Is it possible to include all of the `<NativeAssemblyReference` from the calling BDN project instead of hard-coding them here? That way they only need to be defined in `ML.Benchmarks.csproj`.